hack/create-cluster.sh

```diff
@@ -47,13 +47,44 @@ if [ -f "${KUBECONFIG}" ];then
   fi
 fi
 
-kind create cluster --name "${CLUSTER_NAME}" --kubeconfig="${KUBECONFIG}" --wait=120s
+util::get_macos_ipaddress # Adapt for macOS
+
+# create a cluster to deploy karmada control plane components.
+if [[ -n "${MAC_NIC_IPADDRESS}" ]]; then
+    CONF=`mktemp`
+cat <<EOF >> $CONF
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+networking:
+  apiServerAddress: "${MAC_NIC_IPADDRESS}"
+nodes:
+- role: control-plane
+  image: kindest/node:v1.19.11
+  extraPortMappings:
+  - containerPort: 6443
+    hostPort: 8443
```

Comment:
Maybe it's better to place the config file in dir [artifacts/kindClusterConfig](https://github.com/karmada-io/karmada/tree/master/artifacts/kindClusterConfig)
And shall we only create one cluster by `hack/create-cluster.sh` because the `hostPort: 8843` is fixed.